src/Symfony/Component/Workflow/Registry.php

```diff
@@ -24,17 +23,11 @@ class Registry
     private $workflows = array();
 
     /**
-     * @param Workflow                        $workflow
-     * @param string|SupportStrategyInterface $supportStrategy
+     * @param Workflow                 $workflow
+     * @param SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $supportStrategy)
+    public function add(Workflow $workflow, SupportStrategyInterface $supportStrategy)
```

Comment:
It should be possible to deprecate using something else than a `SupportStrategyInterface` in 3.4 and add this type-hint for 4.0, I don't see the problem 😕